coveo-stew/test_coveo_stew/test_stew_ci.py

```diff
@@ -0,0 +1,357 @@
+import os
+from contextlib import contextmanager
+from typing import Generator, Final
+
+import pytest
+from _pytest.tmpdir import TempPathFactory
```

Comment:
I had to modified some code recently when I bumped `poetry.lock` because we imported something from `_pytest`. Is there anything else we could use that is less likely to break? Why not the usual `tempfile`? I used it quite a bit in [dump-collector](https://github.com/coveo/dump-collector). :thinking: